guacamole-ext/src/test/java/org/apache/guacamole/properties/TimeZoneGuacamolePropertyTest.java

```diff
@@ -0,0 +1,230 @@
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.guacamole.GuacamoleException;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+/**
+ * Tests that validate Time Zone property input.
+ */
+public class TimeZoneGuacamolePropertyTest {
+    
+    /**
+     * An array of valid TimeZones that should be correct parsed by the TimeZone
+     * property, returning either the same or synonymous zone.
+     */
+    private static final List<String> TZ_TEST_VALID = Arrays.asList(
+            "America/Los_Angeles",
+            "America/New_York",
+            "Australia/Sydney",
+            "Africa/Johannesburg",
+            "Asia/Shanghai"
+    );
+    
+    /**
+     * An array of invalid timezone names that should be parsed to GMT, which
+     * should cause an exception to be thrown by the TimeZone property.
+     */
+    private static final List<String> TZ_TEST_INVALID = Arrays.asList(
+            "Chips/Guacamole",
+            "Chips/Queso",
+            "Chips/Salsa",
+            "Mashed/Avacado",
+            "Pico/De_Guayo"
+    );
+    
+    /**
+     * An array of valid GMT specifications that should be correctly parsed
+     * by the TimeZone property as GMT.
+     */
+    private static final List<String> TZ_GMT_VALID = Arrays.asList(
+            "GMT",
+            "GMT-0000",
+            "GMT+000",
+            "GMT+00:00",
+            "GMT-0:00",
+            "GMT+0"
+    );
+    
+    /**
+     * An array of invalid GMT specifications that should cause an exception to
+     * be thrown for the TimeZone property.
+     */
+    private static final List<String> TZ_GMT_INVALID = Arrays.asList(
+            "GMTx0000",
+            "GMT=00:00",
+            "GMT0:00",
+            "GMT+000000",
+            "GMT-000:000",
+            "GMT100"
+    );
+    
+    /**
+     * An array of custom GMT offsets that should evaluate correctly for
+     * the TimeZone property.
+     */
+    private static final List<String> TZ_CUSTOM_VALID = Arrays.asList(
+            "GMT-23:59",
+            "GMT+01:30",
+            "GMT-00:30",
+            "GMT-11:25"
+    );
+    
+    /**
+     * An array of invalid custom GMT offsets that should cause an exception
+     * to be thrown by the TimeZone property.
+     */
+    private static final List<String> TZ_CUSTOM_INVALID = Arrays.asList(
+            "GMT-9999",
+            "GMT+2500",
+            "GMT+29:30",
+            "GMT-1:99",
+            "GMT+10:65"
+    );
+    
+    private static final TimeZoneGuacamoleProperty WHERE_IN_WORLD =
+            new TimeZoneGuacamoleProperty() {
```

Comment:
Please document this very important property. ;)

Reply:
Ah, yes.  Done.